nav2_recoveries/src/recovery_server.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2018 Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include <memory>
+#include <string>
+#include <vector>
+#include "nav2_recoveries/recovery_server.hpp"
+
+namespace recovery_server
+{
+
+RecoveryServer::RecoveryServer()
+: nav2_util::LifecycleNode("nav2_recoveries", "", true),
+  plugin_loader_("nav2_core", "nav2_core::Recovery")
```

Comment:
Why does the loader needs to be a class member and not a local variable in `loadRecoveryPlugins`?

Reply:
To find the plugins, it needs to crawl the filesystem looking for different things so on construction limits us to doing this at most once and crashes immediately if certain inputs are invalid. Additionally I see a time in the not so distant future when we circle back to dynamic reconfigure like behavior that we can to load and unload plugins for tuning and having the member will allow for that.

But a more `this is why I did it` vs a justification reason is that's how I've always done it and that's how all the examples of pluginlib I've seen do it, so I've just never argued with it.

Reply:
Ok, mainly curious. I'm ok leaving it as a member variable.